test/extended/cli/version.go

```diff
@@ -0,0 +1,69 @@
+package cli
+
+import (
+	"regexp"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[sig-cli] oc version", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("oc-version")
+	ocAdmin := oc.AsAdmin()
+
+	g.It("check kubernetes version matches the io.openshift.build.versions label", func() {
+		imageName, err := ocAdmin.Run("get").Args("pods", "-l", "app=kube-controller-manager", "-n", "openshift-kube-controller-manager", "-o=jsonpath={..spec.containers[0].image}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		masterNode, err := ocAdmin.WithoutNamespace().Run("get").Args("nodes", "--selector=node-role.kubernetes.io/master=", "-o=jsonpath={.items[*].metadata.name}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		masterName := strings.Fields(masterNode)
+		imageInfo, err := oc.AsAdmin().Run("debug").Args("node/"+masterName[0], "--", "chroot", "/host", "oc", "image", "info", "--registry-config=/var/lib/kubelet/config.json", strings.Fields(imageName)[0]).Output()
+
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		re1 := regexp.MustCompile(`io.openshift.build.versions=kubernetes=[1-9]{1}.[1-9]{1,2}.[1-9]{1}`)
+		kversion := re1.FindAllString(imageInfo, -1)
+		kversion = strings.Split(kversion[0], "=")
+		kversionNum := kversion[len(kversion)-1]
```

Comment:
@kasturinarra it looks like you'll need to add checks here to ensure that `kversion` has appropriate length and similarly below when you're accessing `result`.